test/data_tests/cassiopeia_tree_test.py

```diff
@@ -0,0 +1,476 @@
+"""
+Tests for the CassiopeiaTree object in the data module.
+"""
+import unittest
+
+import ete3
+import networkx as nx
+import pandas as pd
+
+import cassiopeia as cas
+from cassiopeia.data import utilities as data_utilities
+from cassiopeia.data.CassiopeiaTree import CassiopeiaTreeError
+
+
+class TestCassiopeiaTree(unittest.TestCase):
+    def setUp(self):
+
+        # test_nwk and test_network should both have the same topology
+        self.test_nwk = "((node3,(node7,(node9,(node11,(node13,(node15,(node17,node18)node16)node14)node12)node10)node8)node4)node1,(node5,node6)node2)node0;"
+        self.test_network = nx.DiGraph()
+        self.test_network.add_edges_from(
+            [
+                ("node0", "node1"),
+                ("node0", "node2"),
+                ("node1", "node3"),
+                ("node1", "node4"),
+                ("node2", "node5"),
+                ("node2", "node6"),
+                ("node4", "node7"),
+                ("node4", "node8"),
+                ("node8", "node9"),
+                ("node8", "node10"),
+                ("node10", "node11"),
+                ("node10", "node12"),
+                ("node12", "node13"),
+                ("node12", "node14"),
+                ("node14", "node15"),
+                ("node14", "node16"),
+                ("node16", "node17"),
+                ("node16", "node18"),
+            ]
+        )
+
+        # this should obey PP for easy checking of ancestral states
+        self.character_matrix = pd.DataFrame.from_dict(
+            {
+                "node3": [1, 0, 0, 0, 0, 0, 0, 0],
+                "node7": [1, 1, 0, 0, 0, 0, 0, 0],
+                "node9": [1, 1, 1, 0, 0, 0, 0, 0],
+                "node11": [1, 1, 1, 1, 0, 0, 0, 0],
+                "node13": [1, 1, 1, 1, 1, 0, 0, 0],
+                "node15": [1, 1, 1, 1, 1, 1, 0, 0],
+                "node17": [1, 1, 1, 1, 1, 1, 1, 0],
+                "node18": [1, 1, 1, 1, 1, 1, 1, 1],
+                "node5": [2, 0, 0, 0, 0, 0, 0, 0],
+                "node6": [2, 2, 0, 0, 0, 0, 0, 0],
+            },
+            orient="index",
+        )
+
+    def test_newick_to_networkx(self):
```

Comment:
We can skip test docstrings for now, but we should consider documenting the purpose of each test. (I like to treat tests just like the library code in terms of documentation, because I view tests as the only reliable documentation for how an object should be used.)

Reply:
Great idea! This is going to have to be a larger effort to go back and document all other tests... joy.